### PR TITLE
feat: Adds DevTools extension configuration and methods required by DevTools

### PR DESCRIPTION
**Pull request by vlidholt**
- Adds DevTools extension config
- Adds `loadPasswordsFromMap` method on `PasswordManager`
- Adds tests for `PasswordManager`

### CODE REVIEW

File: packages/serverpod_shared/test/password_manager_test.dart [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  database: 'staging db pass'
+  serviceSecret: 'staging service secret'
+
+production:
+  database: 'production db pass'
+  serviceSecret: 'production service secret'
+''';
+
+void main() {
+  group(
```
Comment (Isakdl) [inline packages/serverpod_shared/test/password_manager_test.dart:25]: Suggestion: 
 * Add a test with an empty passwords file
 * Add a test for a runMode that doesn't exist.

File: packages/serverpod_shared/test/password_manager_test.dart [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      'Given default password config when loading Map in runMode development then',
+      () {
+    const runMode = 'development';
+
+    var passwords = PasswordManager(runMode: runMode)
+        .loadPasswordsFromMap(loadYaml(_defaultPasswordConfig));
+
+    test('Passwords are not null', () {
```
Comment (Isakdl) [inline packages/serverpod_shared/test/password_manager_test.dart:33]: Nit: lower case letter on the first word when you put them in a group:

The log output will be:
`Given default password config when loading Map in runMode development then Passwords are not null`

**Review by Isakdl — CHANGES_REQUESTED** [review]
Looking good! two small comments on the tests.

**Review by Isakdl — APPROVED** [review]
LGTM 🚀
